### ISSUE TITLE
postdrop permission issues

### ISSUE THREAD

**Issue by skibbipl** [firehol/netdata#1582]
Hello,
I've installed netdata (1.4.0) on my server (debian testing) but I constantly get following errors:
`postfix/postdrop[14743]: warning: mail_queue_enter: create file maildrop/761931.14743: Read-only file system`
The error is self-explanatory but I'm not sure what is the best way to fix it. According to `ps` postdrop is running by `netdata` user:
`netdata  14743  0.0  0.1  81124  6636 ?        S    14:21   0:00 /usr/sbin/postdrop -r`
But default permissions for Maildrop directory in debian are:
`drwx-wx--T  2 postfix  postdrop 4096 sty 17 13:45 maildrop`

Any suggestions are welcome. Thanks!

**Comment by ktsaou** [comment]
Sorry.. I don't know.
Can you send an email as any other user (except root)?

**Comment by skibbipl** [comment]
Yes, it works perfectly fine for every user on the system (in fact it's only one user). I used:
`mail -s "Test" test@test.com`

**Comment by ktsaou** [comment]
Hm.. I think you need to send an email from another non-root user. The test you made sent the email from which user?

**Comment by skibbipl** [comment]
The server configuration is pretty simple:
- postfix is the MTA - in fact it works as SMTP server for dedicated domain.
- admin is the only non-root user
I can send mails as admin with `mail` command. Why netdata uses postdrop?

**Comment by ktsaou** [comment]
> Why netdata uses postdrop

it does not. It sends emails using the `sendmail` command. All MTAs provide a `sendmail` compatible interface, which is the most commonly used interface to send emails from applications.

> admin is the only non-root user

Your system should have at least a dozen more users, used by various applications (like user `netdata`). Check your `/etc/passwd`. 

You can be become any of them with this:

```sh
# to become user, run as root
su -s /bin/sh USERNAME

# now try to send an email
```

**Comment by skibbipl** [comment]
I can send mails as any user in the system.
```
Jan 19 15:46:54 server postfix/pickup[29379]: 00CCBE10DC: uid=126 from=<netdata>
Jan 19 15:46:54 server postfix/cleanup[5714]: 00CCBE10DC: message-id=<20170119144654.00CCBE10DC@domain.tld>
Jan 19 15:46:54 server postfix/qmgr[13477]: 00CCBE10DC: from=<netdata@domain.tld>, size=307, nrcpt=1 (queue active)
Jan 19 15:46:54 server dovecot: lda(root): sieve: msgid=<20170119144654.00CCBE10DC@domain.tld>: stored mail into mailbox 'Server logs'
Jan 19 15:46:54 server postfix/local[5716]: 00CCBE10DC: to=<root@domain.tld>, orig_to=<root>, relay=local, delay=0.26, delays=0.14/0.01/0/0.11, dsn=2.0.0, status=sent (delivered to command: /usr/lib/dovecot/dovecot-lda -f "$SENDER"
 -a "$RECIPIENT")
```

**Comment by ktsaou** [comment]
hm... while running as user `netdata` could you please run:

```sh
/usr/libexec/netdata/plugins.d/alarm-notify.sh test
```

**Comment by skibbipl** [comment]
Result:
```
$ /usr/lib/x86_64-linux-gnu/netdata/plugins.d/alarm-notify.sh test
/usr/lib/x86_64-linux-gnu/netdata/plugins.d/alarm-notify.sh: not sending notification for  on '.'
```

**Comment by ktsaou** [comment]
hm... there is no `sendmail` in you system.

Well, could you please run `ps fax` when this happens? Is `postdrop` in the process tree of netdata?

Search also for `postdrop` is the netdata source. There isn't any reference to it.

So, something else is trying to send an email as user netdata...

**Comment by skibbipl** [comment]
No, I have sendmail on my server. Here is process tree:
```
netdata  28354  0.4  0.9 150532 38408 ?        Ssl  06:25   1:05 /usr/sbin/netdata -D
netdata  28369  0.4  0.4 104384 19352 ?        Sl   06:25   1:11  \_ /usr/bin/python /usr/lib/x86_64-linux-gnu/netdata/plugins.d/python.d.plugin 1
netdata  28716  0.0  0.0  12704  3372 ?        S    06:26   0:00  \_ bash /usr/lib/x86_64-linux-gnu/netdata/plugins.d/alarm-notify.sh sysadmin server 1484890058 1484889981 2 1484889955 disk_space_usage disk_space.dm-1 /home WARNING UNINIT...
netdata  28721  0.0  0.1  81136  6552 ?        S    06:26   0:00  |   \_ /usr/sbin/sendmail -t
netdata  28722  0.0  0.1  81128  6480 ?        S    06:26   0:00  |       \_ /usr/sbin/postdrop -r
netdata  19405  1.0  0.0  16228  3004 ?        S    10:25   0:12  \_ /usr/lib/x86_64-linux-gnu/netdata/plugins.d/apps.plugin 1
netdata  19407  0.0  0.0  12536  3212 ?        S    10:25   0:00  \_ bash /usr/lib/x86_64-linux-gnu/netdata/plugins.d/tc-qos-helper.sh 1

```

**Comment by ktsaou** [comment]
hm... I really don't know how to help you... As you can see sendmail is running postdrop. `postdrop` is supposed to work for all users.

@philwhineray any ideas?

**Comment by philwhineray** [comment]
I don't know how much help this will be, anyway... The error from  postdrop itself suggests a missing setuid on the binary.

However, it looks from the logs like @skibbipl has postfix configured to use dovecot to deliver mail under working conditions, and if so that should not be needed, since dovecot should do the work.

If dovecot is normally used, problems with maildrop wouldn't be seen, the question is why maildrop is used in this case.

I would start by checking the source and destination email are as expected from netdata.

If that's OK, then verify the postfix config. Either source, destination or the combination seems to be matching a different case than the manual tests that have been run.

**Comment by ktsaou** [comment]
@skibbipl have you solved the issue?

I am closing this, since it seems not related to netdata. If however you believe there is a fault in netdata, or you need help, just post.

**Comment by skibbipl** [comment]
@ktsaou Unfortunately not.
But I'm pretty sure that issue is related to netdata, as it calls postdrop according to process tree.
I've filled debian bug, just in case the issue is related to bad packaging.

**Comment by ktsaou** [comment]
ok. I am not sure how to help though. netdata is just calling `sendmail`. If the debian folks suggest we should do something, let us know.

**Comment by skibbipl** [comment]
I'll try to install latest netdata from sources to make sure that I use official code. Will give feedback soon.

**Comment by FedericoCeratto** [comment]
Probably related to https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=851852 - @skibbipl

**Comment by skibbipl** [comment]
After upgrading to 1.5 on one of the servers the issue is gone.

**Comment by skibbipl** [comment]
Hello,
Yesterday I installed latest netdata 1.18 from Debian packages and I'm again seeing this behaviour.
@FedericoCeratto I guess the issue is similar to 851852 debian bug. Perhaps systemd hardening options are too strict?

**Comment by putnam** [comment]
This happened to me today again with netdata 1.19.0-3 on Debian testing. After unattended-upgrades ran, it overwrote my modified systemd config file where I added the line suggested above.

But this time, netdata completely hung. It is stuck waiting to send that alert email. The web interface will accept TCP connections and then hang without responding. Only once I re-enabled r/w on the maildrop queue did netdata behave itself again.

Clearly there is a Debian issue here, but I also am not sure netdata should be blocking on all connections while waiting to send an email.

**Comment by PhantomGhost** [comment]
This same problem is also happening on Ubuntu 20.04 beta, Netdata runs, but doesn't respond at all, its just hung, and I have the same error.

**Comment by Compizfox** [comment]
Same issue on Debian Bullseye, but @skibbipl's suggestion works.

**Comment by prologic** [comment]
Guys as this issue is quite old and things have changed quite a bit since  then I'd appreciate it if this is still an issue for anyone to please open a new issue. Thanks!

**Comment by mikkorantalainen** [comment]
If somebody is still having this issue, start with `sudo postfix check` to run postfix self-test. You could have incorrect permissions or e.g. missing setgid bit on /usr/sbin/maildrop.
